azurerm/internal/services/network/resource_arm_application_gateway.go

```diff
@@ -660,9 +661,8 @@ func resourceArmApplicationGateway() *schema.Resource {
 						},
 
 						"capacity": {
-							Type:         schema.TypeInt,
-							Optional:     true,
-							ValidateFunc: validation.IntBetween(1, 32),
+							Type:     schema.TypeInt,
```

Comment:
The doc's say that this is a required attribute but is optional only if the `autoscale_configuration` attribute is set to `true`. Do you think we can add an additional check to the `CustomizeDiff` to check this condition? I think this may have been overlooked with the initial implementation of this resource.

Reply:
I think your concern is reasonable. I just add the validation in the CustomizeDiff function. Please have a look. Thanks.